packages/storage-web/src/index.ts

```diff
@@ -5,3 +5,98 @@
  * LICENSE file in the root directory of this source tree.
  *
  */
+import {
+  IStorageBackend,
+  EmptyStorageModel,
+  StorageOptions,
+} from '@react-native-community/async-storage';
+
+class WebStorage<T extends EmptyStorageModel = EmptyStorageModel>
+  implements IStorageBackend<T> {
+  storage: any;
+  constructor(sessionStorage: boolean | string = false) {
+    this.storage = sessionStorage ? window.sessionStorage : window.localStorage;
+  }
+
+  async getSingle<K extends keyof T>(
+    key: K,
+    opts?: StorageOptions,
+  ): Promise<T[K] | null> {
+    if (opts) {
+      // noop
+    }
+    return this.storage.getItem(key);
+  }
+
+  async setSingle<K extends keyof T>(
+    key: K,
+    value: T[K],
+    opts?: StorageOptions,
+  ): Promise<void> {
+    if (opts && !opts.replaceCurrent) {
+      const current = this.storage.getItem(key);
+      if (!current) {
+        this.storage.setItem(key, value);
+      }
+      return;
+    }
+    return this.storage.setItem(key, value);
+  }
+
+  async getMany<K extends keyof T>(
+    keys: Array<K>,
+    opts?: StorageOptions,
+  ): Promise<{[k in K]: T[k] | null}> {
+    if (opts) {
+      // noop
+    }
+    return Promise.all(keys.map(k => this.storage.getItem(k)));
+  }
+
+  async setMany<K extends keyof T>(
+    values: Array<{[k in K]: T[k]}>,
```

Comment:
```suggestion
    values: Array<Partial<{[k in K]: T[k]}>>,,
```

Otherwise, you'll be expected to pass full model every time